Automattic/simplenote-android
# Refactor tag screens to filter out collaborators

Fixes #1480

### Fix

This PR makes a series of fixes to make the tag screens consistent with the dedicated UI for collaboration added in #1476. The changes are the following:

- Left panel (navigation drawer) filters out tags that are collaborators (emails)

| Before      | After |
| ----------- | ----------- |
| ![](https://d.pr/i/ZW14eJ+.jpg)      | ![](https://d.pr/i/2AMksx+.jpg)       |

- The list of tags (in the Edit Tags screen) filters out tags that are collaborators (emails)  

| Before      | After |
| ----------- | ----------- |
| ![](https://d.pr/i/3XJUBU+.jpg)      | ![](https://d.pr/i/HbIxq6+.jpg)       |



- If you try to add an email as a tag from the list of tags (in the Edit Tags screen), it shows an error

| Before      | After |
| ----------- | ----------- |
| ![](https://d.pr/i/WHYWgw+.jpg)      | ![](https://d.pr/i/KHmTF3+.jpg)       |



- If you try to edit a tag as an email from the list of tags (in the Edit Tags screen), it shows an error

| Before      | After |
| ----------- | ----------- |
| ![](https://d.pr/i/V9bafO+.jpg)      | ![](https://d.pr/i/mIXQm4+.jpg)       |



### Test

1. Go to a note
2. Tap on the tags field at the bottom and add a couple of tags
3. Tap on (...) button at the top right corner and tap on Collaborators
4. Add a couple of collaborators
5. Go back to edit the note
6. Go back to the list of notes
7. Open the left panel. Just the tags and not the emails should be shown
8. Tap on one of the tags and the list of notes should be filtered by the notes that have that tag.
9. Tap on All Note and the list of notes should be filtered to all notes
8. Tap on `EDIT` button on the left panel
9. Just the tags and not the emails should be shown
10. Try to add a tag
11. Try to add an email on the `Add Tag` form. It should show an error
12. Go back and dismiss the dialog
13. Tap on one of the tags on the list
14. Try to edit the tag and write it as an email on the `Edit Tag` form. It should show an error
12. Go back and dismiss the dialog

### Review

Only one developer and one designer are required to review these changes, but anyone can perform the review.


### Release

`RELEASE-NOTES.txt` was updated in  aa5c545f2c1bd35deebdb2c56cb8154c2cb5bf93 with:
> [Internal] Updated list of tags to filter out collaborators (emails)
